worlds/tunic/__init__.py

```diff
@@ -80,28 +83,34 @@ class TunicWorld(World):
     tunic_portal_pairs: Dict[str, str]
     er_portal_hints: Dict[int, str]
     seed_groups: Dict[str, SeedGroup] = {}
-    shop_num: int = 1  # need to make it so that you can walk out of shops, but also that they aren't all connected
+    used_shop_numbers: Set[int]
     er_regions: Dict[str, RegionInfo]  # absolutely needed so outlet regions work
 
     def generate_early(self) -> None:
-        if self.options.logic_rules >= LogicRules.option_no_major_glitches:
-            self.options.laurels_zips.value = LaurelsZips.option_true
-            self.options.ice_grappling.value = IceGrappling.option_medium
-            if self.options.logic_rules.value == LogicRules.option_unrestricted:
-                self.options.ladder_storage.value = LadderStorage.option_medium
-
         self.er_regions = tunic_er_regions.copy()
         if self.options.plando_connections:
             for index, cxn in enumerate(self.options.plando_connections):
-                # making shops second to simplify other things later
-                if cxn.entrance.startswith("Shop"):
-                    replacement = PlandoConnection(cxn.exit, "Shop Portal", "both")
+                # flip any that are pointing to exit to point to entrance so that I don't have to deal with it
+                if self.options.decoupled and cxn.direction == "exit":
+                    replacement = PlandoConnection(cxn.exit, cxn.entrance, "entrance", cxn.percentage)
+                    self.options.plando_connections.value.remove(cxn)
+                    self.options.plando_connections.value.insert(index, replacement)
```

Comment:
More of a nit but this feels like something that could be simplified a little if you shifted some variables around 
```suggestion
            replacement = None
            if self.options.decoupled:
                if cxn.direction == "exit":
                    replacement = PlandoConnection(cxn.exit, cxn.entrance, "entrance", cxn.percentage)
                elif cxn.direction == "both" and cxn.entrance == cxn.exit:
                    replacement = PlandoConnection(cxn.entrance, cxn.exit, "entrance")
            elif cxn.direction != "both":
                replacement = PlandoConnection(cxn.entrance, cxn.exit, "both", cxn.percentage)

            if replacement:
                replace_connection(index, cxn, replacement)
```

Reply:
```py
    def replace_connection(index, old_cxn, new_cxn):
        self.options.plando_connections.value.remove(old_cxn)
        self.options.plando_connections.value.insert(index, new_cxn)
```

Reply:
Definitely implementing part of this o: I'm leaving the different `if`s separate though, mainly so it's easier for me to read and interpret my intent